### PR TITLE
EVO-1099: fix embedded documents

### PR DESCRIPTION
**Pull request by narcoticfresh**
this was a heated topic and a long journey to get here. this one replaces #314 to have a cleaner git history. i'm sorry for the stolen commits guys - I tried like 10 times to rebase but it had so many conflicts that I decided it's easier to start in a new branch, reset to develop and re-commit all changes in a consolidated fashion.

this is an important change that we need for graviton 1.0 and the first client delivery.. only with this PR, we finally handle saving and handling of all relations (be it ref or embed) correctly on the MongoDB level.

it introduces
* changes to GeneratorBundle to generate an *Embedded version for every document alongside the normal Document class (with all that comes with it - meaning also repos, services and so on)
* correctly generate the services. If now a service has a field defined like "field.subfield", the class FieldEmbedded is used for that hash, containing field "subfield"
* Some Bundles (Person and File) used some wrong classes that also get corrected.

testing
* Since the beginning of this feature, we had an initial test (`EmbeddingDocumentsTest`) from @mrix that demonstrated the wrong behavior. We worked until this test was all good, what it is now. This test now is part of our test suite; the test service definitions for that are now part of the `TestBundle`.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
         "libgraviton/codesniffer": "~1.3",
         "lapistano/proxy-object": "dev-master",
         "johnkary/phpunit-speedtrap": "~1.0",
-        "graviton/test-services-bundle": "^0"
+        "graviton/test-services-bundle": "*"
```
Comment (lapistano) [inline composer.json]: this will checkout 'dev-develop' is this ok? Why not tag a new version?

Comment (narcoticfresh) [inline composer.json]: hm, it shouldn't.. `*` means "any version", meaning a tag..
the composer.lock also looks like that, 
```

Comment (lapistano) [inline composer.json]: ok. I guess I was confused by the 'minimum-stability: dev' configuration in the configuration.

File: src/Graviton/GeneratorBundle/Resources/skeleton/document/DocumentBase.php.twig [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+use JMS\Serializer\DeserializationContext;
+
+{% set translatableFields = [] %}
+{% set preTranslatedFields = [] %}
+
+/**
+ * {{ base }}Document\{{ document }}Base
+ *
+ * @author   List of contributors <https://github.com/libgraviton/graviton/graphs/contributors>
+* @license  http://opensource.org/licenses/gpl-license.php GNU Public License
```
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Resources/skeleton/document/DocumentBase.php.twig]: indentation

File: src/Graviton/GeneratorBundle/Resources/skeleton/document/DocumentEmbedded.php.twig [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * document class for {{ base }}Document\{{ document }}Embedded
+ */
+
+namespace {{ base }}Document;
+
+/**
+ * @author   List of contributors <https://github.com/libgraviton/graviton/graphs/contributors>
+* @license  http://opensource.org/licenses/gpl-license.php GNU Public License
```
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Resources/skeleton/document/DocumentEmbedded.php.twig]: indentation (every docblock in file)

**Pull request merged**